nbgrader/tests/apps/test_nbgrader_autograde.py

```diff
@@ -568,6 +568,72 @@ def test_update_newer_single_notebook(self, course_dir):
         assert p1 != self._file_contents(join(course_dir, "autograded", "foo", "ps1", "p1.ipynb"))
         assert p2 == self._file_contents(join(course_dir, "autograded", "foo", "ps1", "p2.ipynb"))
 
+    def test_hidden_tests_single_notebook(self, db, course_dir):
+        with open("nbgrader_config.py", "a") as fh:
+            fh.write("""c.NbGrader.db_assignments = [dict(name='ps1', duedate='2015-02-02 14:58:23.948203 PST')]\n""")
+            fh.write("""c.NbGrader.db_students = [dict(id="foo"), dict(id="bar")]""")
+
+        self._copy_file(
+            join("files", "test.ipynb"),
+            join(course_dir, "source", "ps1", "p1.ipynb")
+        )
+        self._copy_file(
+            join("files", "test-hidden-tests.ipynb"),
+            join(course_dir, "source", "ps1", "p2.ipynb")
+        )
+        self._copy_file(
+            join("files", "test-hidden-tests.ipynb"),
+            join(course_dir, "source", "ps1", "p3.ipynb")
+        )
+        run_nbgrader(["assign", "ps1", "--db", db])
+
+        # make sure hidden tests are removed in release
+        with open(join(course_dir, "release", "ps1", "p2.ipynb"), 'r') as nb:
+            source = nb.read()
+        assert "BEGIN HIDDEN TESTS" not in source
+
+        self._copy_file(
```

Comment:
This is buggy test -- it should copy the files from the release directory, not from the original `files` directory.